src/v3/src/util/settingsUtils.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import union from 'lodash/union';
+
+import BrowserFeatures from '../../../util/BrowserFeatures';
+import config from '../../../config/config.json';
+import Util from '../../../util/Util';
+
+import { WidgetProps } from '../types';
+import { LanguageCode } from '../../../types';
+
+export const getSupportedLanguages = (widgetProps: WidgetProps): string[] => {
+  const { i18n, language, assets: { languages } = {} } = widgetProps;
+  const supportedLanguages = languages || config.supportedLanguages;
+  const customLanguages = Object.keys(i18n || {});
+
+  return union(
+    supportedLanguages,
+    customLanguages,
+    typeof language === 'string' ? [language] : [],
+  );
+};
+
+export const getLanguageCode = (widgetProps: WidgetProps): string => {
+  const { language } = widgetProps;
+  const supportedLanguages = getSupportedLanguages(widgetProps);
+  const userLanguages = BrowserFeatures.getUserLanguages().map((lang: string) => {
+    if (lang === 'nl') {
+      return 'nl-NL';
+    }
+    if (lang === 'pt') {
+      return 'pt-BR';
+    }
+    return lang;
+  });
+  
+  const preferredLanguages = [...userLanguages];
+  const supportedLangsLowercase = Util.toLower(supportedLanguages);
+
+  // Any developer defined "language" takes highest priority:
```

Comment:
come from v2? possible to refactor v2 code into some exportable function, then reuse here?

Reply:
in v2 this is done in the Settings.ts file, I am not entirely sure how we can refactor that or if we should. Any advice?